File: src/main/java/com/example/RESTAPIarticle/entity/ArticleContent.java
package com.example.RESTAPIarticle.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.NoArgsConstructor;
import org.springframework.stereotype.Component;

import javax.persistence.*;

@Entity
@Table(name="Content")
@Component
@NoArgsConstructor
public class ArticleContent {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    @Column(name="id")
    private int id;

    @Column(name="title")
    private String title;

    @Column(name="text")
    private String text;

    @OneToOne(mappedBy = "content")
    @JsonIgnore
    private Article article;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public Article getArticle() {
        return article;
    }

    public void setArticle(Article article) {
        this.article = article;
    }

    @Override
    public String toString() {
        return "ArticleContent{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", text='" + text + '\'' +
                ", article=" + article +
                '}';
    }
}

File: src/main/java/com/example/articles/entity/ArticleContent.kt
package com.example.articles.entity

import org.hibernate.Hibernate
import javax.persistence.*

@Entity
@Table(name = "Content")
data class ArticleContent(
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id")
    val id: Int,

    @Column(name = "title")
    val title: String,

    @Column(name = "text")
    val text: String,

    @OneToOne(mappedBy = "content")
    val article: ArticlePost?,
) {
    constructor() : this(
        0,
        "asd",
        "asd",
        null
    )

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
        other as ArticleContent

        return id == other.id
    }

    override fun hashCode(): Int = hashCode()

    @Override
    override fun toString(): String {
        return this::class.simpleName + "(id = $id , title = $title , text = $text , article = $article )"
    }
}